### ISSUE TITLE
Make bootcoldist() more flexible regarding vismodeldata input

### ISSUE THREAD

**Issue by Bisaloo** [rmaia/pavo#195]
According to the docs, `vismodeldata` can be a df with independently computed qcatches:

https://github.com/rmaia/pavo/blob/db63e94ccde5bfa9219422ef08488d259a354158/R/bootcoldist.R#L6-L9

Yet, the argument check prevents the use from anything else than `colspace` or `vismodel` objects:

https://github.com/rmaia/pavo/blob/db63e94ccde5bfa9219422ef08488d259a354158/R/bootcoldist.R#L42-L44

This is also not coherent with the fact the `coldist()` does accept df with independently computed qcatches.

**Comment by Bisaloo** [comment]
Temporary workaround in case someone needs it:
``` r
library(pavo)
#> Welcome to pavo 2! Take a look at the latest features (and update your bibliography) in our recent publication: Maia R., Gruson H., Endler J. A., White T. E. (2019) pavo 2: new tools for the spectral and spatial analysis of colour in R. Methods in Ecology and Evolution, 10, 1097-1107.

data(sicalis)

vis_sicalis <- vismodel(sicalis)

gr <- gsub("ind..", "", rownames(vis_sicalis))

# Strip vismodel class attribute from 'sicalis' to mimic a data.frame with 
# independently computed quantum catches
class(vis_sicalis) <- "data.frame"

# will fail
bootcoldist(vis_sicalis, by = gr, n = c(1, 2, 2, 4), weber = 0.1, weber.achro = 0.1)
#> Error: object must be a "vismodel" or "colspace" result

# Trick bootcoldist() into thinking our dataset is indeed the output of 
# vismodel()
class(vis_sicalis) <- c("vismodel", "data.frame")

bootcoldist(vis_sicalis, by = gr, n = c(1, 2, 2, 4), weber = 0.1, weber.achro = 0.1)
#> Warning: Quantum catch are relative, distances may not be meaningful
#> Calculating noise-weighted Euclidean distances
#>      dS.mean    dS.lwr   dS.upr
#> B-C 4.626548 3.0982122 7.065537
#> B-T 1.741764 0.2231299 5.314494
#> C-T 6.110519 4.0002001 8.690730
```

<sup>Created on 2020-04-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by thomased** [comment]
So was this solved with 97500d9bc6bef5c1f27d10a5250a7533b6cd3c58?

**Comment by Bisaloo** [comment]
Oh right, I forgot. Thanks for the heads up!
